exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -113,6 +113,7 @@ class PrometheusMetricReader(MetricReader):
     """
 
     def __init__(self, prefix: str = "") -> None:
+
```

Comment:
Looks like if someone sets the OTLP environemnt variable to `DELTA`, it will cause this to be configured for deltas:

https://github.com/open-telemetry/opentelemetry-python/blob/078eb655d2c564a7f39bbb8511c68718834ce172/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py#L184

is that a separate issue and we should remove it from the base class? Could also just explicitly pass CUMULATIVE for each instrument kind to the parent constructor to fix this

Reply:
The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view) says:

> The aggregation (optional) to be used. If not provided, the SDK MUST apply a default aggregation configurable on the basis of instrument kind according to the MetricReader instance.

In my opinion, that indicates the aggregation for every instrument must be configurable in the `MetricReader` (which we do already). So I think the `preferred_temporality` should stay in the base `MetricReader` class.

The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/prometheus.md#opentelemetry-metrics-exporter---prometheus) also says:

> A Prometheus Exporter MUST only support Cumulative Temporality.

I see using `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` and `preferred_temporality` as "equivalent" in the sense that they are 2 different mechanisms to configure the same thing. Because of that I would prefer that the Prometheus reader treats them in the same way: regardless of what the configuration mechanism says, every instrument type will always be associated with a cumulative temporality.

We have already decided not to have `preferred_temporality` configuration mechanism in the arguments of the Prometheus Metric Reader constructor. Since we can't remove the `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` configuration mechanism for the Prometheus metric reader, what we can do is to pass a dictionary with only cumulative temporalities to the parent as @aabmass suggests.

We can check the value of `OTEL_EXPORTER_METRICS_TEMPORALITY_PREFERENCE` and if we find a non-cumulative temporality there we can log a warning to let the user know we are overriding a temporality.

Reply:
> In my opinion, that indicates the aggregation for every instrument must be configurable in the `MetricReader` (which we do already). So I think the `preferred_temporality` should stay in the base `MetricReader` class.

Right i'm not suggesting to remove that. Im saying why do we read the **OTLP** specific environment variable in the base MetricReader? Is that a bug or was there some reason for this?

> We can check the value of `OTEL_EXPORTER_METRICS_TEMPORALITY_PREFERENCE` and if we find a non-cumulative temporality there we can log a warning to let the user know we are overriding a temporality.

That seems reasonable to me

Reply:
The init args take preference over the env. Since we are going to explicitly pass the dict with cumulative this shouldn't be an issue?

Reply:
I prefer to open a new issue for this so that we can move forward with RC2, opened #2792 for that. @aabmass are you ok with addressing this there?

Reply:
Yup

> Since we are going to explicitly pass the dict with cumulative this shouldn't be an issue?

@ocelotl please do this ^^ and otherwise LGTM.

Reply:
Fixed